pkg/scheduler/plugins/coscheduling/core/gang_marshal.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type GangMarshal struct {
+	Name                     string
+	WaitTime                 time.Duration
+	CreateTime               time.Time
+	Mode                     string
+	MinRequiredNumber        int
+	TotalChildrenNum         int
+	GangGroup                []string
+	Children                 sets.String
+	WaitingForBindChildren   sets.String
+	BoundChildren            sets.String
+	OnceResourceSatisfied    bool
```

Comment:
Consider making these names easier to read. ResourceSatisfied?

Reply:
OnceResourceSatisfied means has been satisfied once in history, cause when satified then pod is evicted